biomage/unstage/unstage.py

```diff
@@ -1,23 +1,194 @@
+    click.echo("Removing staging records from DynamoDB")
+    dynamodb = boto3.client("dynamodb")
+    staged_experiments = dynamodb.scan(
+        TableName=config["staging-experiments-table"],
+        ProjectionExpression="experimentId",
+        FilterExpression="begins_with(experimentId, :sandbox_id)",
+        ExpressionAttributeValues={":sandbox_id": {"S": sandbox_id}},
+    )
+
+    staged_experiments = [
+        experiment_id["experimentId"]["S"]
+        for experiment_id in staged_experiments.get("Items")
+    ]
+
+    if len(staged_experiments) == 0:
+        click.echo(
+            f"No scoped experiments registered in table {config['staging-experiments-table']}"
+        )
+        click.echo()
+        return
+
+    all_staging_tables = [
+        table
+        for table in dynamodb.list_tables().get("TableNames")
+        if re.match(".*-staging", table)
+    ]
+
+    # Generate records and create multiple
+    for table in all_staging_tables:
+        records_to_delete = {table: []}
+
+        # Deleting records from DynamoDB requires the primaryKey.
+        # if the table's PK is made up of partitionKey & sortKey,
+        # will have to iterate over all the sortKeys to add as delete parameter
+        key_schema = (
+            dynamodb.describe_table(TableName=table).get("Table").get("KeySchema")
+        )
+
+        if len(key_schema) > 1:
+            sort_key = key_schema[1].get("AttributeName")
+            for experiment_id in staged_experiments:
+
+                # Query table for all keys
+                items_to_delete = dynamodb.query(
+                    TableName=table,
+                    ProjectionExpression=f"experimentId, {sort_key}",
+                    KeyConditionExpression="experimentId = :experiment_id",
+                    ExpressionAttributeValues={":experiment_id": {"S": experiment_id}},
+                ).get("Items")
+
+                if len(items_to_delete) == 0:
+                    break
+
+                for item in items_to_delete:
+                    records_to_delete[table].append(
+                        {
+                            "DeleteRequest": {
+                                "Key": {
+                                    "experimentId": {
+                                        "S": experiment_id,
+                                    },
+                                    sort_key: item[sort_key],
+                                }
+                            }
+                        }
+                    )
+
+        else:
+            records_to_delete[table] = [
+                {"DeleteRequest": {"Key": {"experimentId": {"S": experiment_id}}}}
+                for experiment_id in staged_experiments
+            ]
+
+        if len(records_to_delete[table]) == 0:
+            click.echo(f"No records to delete in table {table}, skipping table...")
+            continue
+
+        try:
+            dynamodb.batch_write_item(RequestItems=records_to_delete)
```

Comment:
why are you doing this `batch_write_item` stuff to delete? You can  just use `delete_item` see docs [here](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.delete_item). Using `delete_item` also will remove the need to create does `records_to_delete` dictionaries which are very error-prone (because they have hardcoded keys instead of relying on the api method `delete_item`).

Reply:
I use it to reduce minimize the request to DynamoDB, so multiple items can be deleted through 1 request

Reply:
But it's right, there is merit to having simpler code. I'm fine with both, better to have it using `delete_item`?

Reply:
I think it's better to change it but I'm not a strong opinion. You decide. Here's my thinking, using batch delete we will have less requests to DynamoDB which is good, however the number of requests in these situations won't be too big so probably does not need optimization. On the other hand, those hard-coded write-item dictionaries are much more probably to have changes in format or induce into error (misspelled keys or similar) than it's counterpart which has named parameters (so at most you can put a wrong value, not key).

Reply:
Hmm.. ok. I'll take a look at the changes required before deciding. Will let you know.

Reply:
Changed to `delete_item` as it's easier to see the API calls